esrc/rescueoverview.php

```diff
@@ -473,4 +483,86 @@ function displayNotes($row, $isCoord = 0, $isSARAgent = 0)
 		}
 	}
 }
+
+/**
+ * Check killboard and update rescue status if activity detected
+ * @param unknown $data
+ */
+function checkKbActivity($data) {
+
+	// create a new database connection
+	$database = new Database();
+	$kb = new KillboardActivity($database);
+	
+	$lastCheck = strtotime($kb->getLastKillboardCheck());
+	$yesterday = strtotime("-1 day");
+	
+	$isNewCheckRequired = $yesterday > $lastCheck;
+	
+	if (!$isNewCheckRequired) {		
+		return;
+	}
+	
+	$kb->updateLastKillboardCheck();
+
+	$rescue = new Rescue($database);
+	
+	foreach ($data as $row) {
+	
+		// have to wait one day because we don't have access to exact time SAR was opened
+		$since = date("Ymd0000", strtotime($row['lastcontact'] . '+1 day'));
+		$characterId = getCharacterId($row['pilot']);
+		
+		if (isKillboardActive($characterId, $since)) {				
+			$rescue->setStatus($row['id'], 'check-killboard');
+		}
+	}
+}
+
+/**
+ * Checks for activity on zkillboard.com
+ * @url string
+ */
+ function getJson($url) {
+	
+	$header = array(
+	    'Content-Type: application/json',
+	    'Accept: application/json'
+	);
+
+	$cURL = curl_init();
+	curl_setopt($cURL, CURLOPT_URL, $url);
+	curl_setopt($cURL, CURLOPT_HTTPGET, true);
+	curl_setopt($cURL, CURLOPT_HTTPHEADER, $header);
+	curl_setopt($cURL, CURLOPT_RETURNTRANSFER, true);
+	$result = curl_exec($cURL);
+	curl_close($cURL);
+
+	return json_decode($result, true);
+}
+
+/**
+ * Gets character ID from eve api
+ 
+ */
+function getCharacterId($pilotName) {
+
+	$url = 'https://esi.evetech.net/latest/search/'
+			. '?categories=character&datasource=tranquility&language=en-us&strict=true&search=' 
+			. urlencode($pilotName);
+	
+	return getJson($url)['character'][0];
+}
+
+/**
+ * Checks for activity on zkillboard.com
+ * @characterId string the ESI character id
+ * @startTime the time/date to search killboard from
+ */
+function isKillboardActive($characterId, $startTime) {
+	$urlKills = 'https://zkillboard.com/api/characterID/' . $characterId . '/startTime/' . $startTime . '/';
+	$activityCount = count(getJson($urlKills));
```

Comment:
This would need some error checking if the zkill API changes/an error is caused. This would help to identify issues.

Reply:
zKill has been known to change its api without warning. perhjaps a year ago, it changed the names of its parameters, for example StartTime becomes startTime.  In that case the API would return:

{"error":"StartTime is an invalid parameter"}